client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -0,0 +1,690 @@
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    return await fetchJson<MatchmakingTimeJson>(
+      `/api/1/matchmaking-times/${encodeURIComponent(type)}`,
+      {
+        method: 'post',
+        body: encodeBodyAsParams<AddMatchmakingTimeRequest>({ startDate, enabled }),
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function deleteMatchmakingTime(
+  type: MatchmakingType,
+  id: number,
+  spec: RequestHandlingSpec,
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    await fetchJson(`/api/1/matchmaking-times/${encodeURIComponent(id)}`, {
+      method: 'delete',
+      signal: spec.signal,
+    })
+  })
+}
+
+const Container = styled(CenteredContentContainer)`
+  padding-top: 16px;
+  padding-bottom: 8px;
+`
+
+const PageHeadline = styled.div`
+  ${titleLarge};
+  margin-top: 16px;
+  margin-bottom: 8px;
+`
+
+const ErrorText = styled.div`
+  ${bodyLarge};
+  color: var(--theme-error);
+`
+
+const AddNewCard = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  padding: 16px;
+  background-color: rgba(255, 255, 255, 0.03);
+  border-radius: 8px;
+  margin-bottom: 16px;
+`
+
+const DateInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 16px;
+`
+
+const DateInput = styled.input`
+  color: var(--theme-on-surface);
+  padding: 8px 12px;
+  border-radius: 4px;
+  border: 1px solid rgba(255, 255, 255, 0.1);
+  background-color: rgba(255, 255, 255, 0.05);
+  font-size: 14px;
+  min-width: 200px;
+
+  &:focus {
+    outline: none;
+    border-color: rgba(255, 255, 255, 0.2);
+    background-color: rgba(255, 255, 255, 0.08);
+  }
+
+  &::-webkit-calendar-picker-indicator {
+    filter: invert(1);
+    opacity: 0.7;
+    cursor: pointer;
+
+    &:hover {
+      opacity: 1;
+    }
+  }
+`
+
+const InvalidDateInput = styled.span`
+  ${bodyMedium};
+  color: var(--theme-error);
+`
+
+const ValidDateIcon = styledWithAttrs(MaterialIcon, { icon: 'check_circle' })`
+  color: var(--theme-success);
+`
+
+const AddNewButton = styled(ElevatedButton)`
+  align-self: flex-start;
+`
+
+const HistoryContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+`
+
+const FutureTimesContainer = styled.div`
+  display: flex;
+  flex-direction: column-reverse;
+  gap: 8px;
+`
+
+const SectionTitle = styled.h3`
+  margin-top: 16px;
+  margin-bottom: 8px;
+  color: var(--theme-on-surface-variant);
+  font-size: 14px;
+  font-weight: 500;
+  text-transform: uppercase;
+  letter-spacing: 0.5px;
+`
+
+const HistoryCard = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 16px;
+  background-color: rgba(255, 255, 255, 0.05);
+  border-radius: 8px;
+  transition: background-color 0.2s ease;
+
+  &:hover {
+    background-color: rgba(255, 255, 255, 0.08);
+  }
+`
+
+const CurrentMatchmakingCard = styled(HistoryCard)`
+  background-color: rgba(255, 193, 7, 0.1);
```

Comment:
I don't know what these colors are but please use stuff from the theme and/or `containerStyles`